ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -345,6 +363,28 @@ private void setTableScanOrElseThrow(final Supplier<KsqlException> exceptionSupp
     }
   }
 
+  private final class NonColumnRefValidator extends TraversalExpressionVisitor<Object> {
```

Comment:
```suggestion
  private final class HasColumnRef extends TraversalExpressionVisitor<Object> {
```
It was really tough to grok `nonColumnRefUnresolvable` - it looks like what this does is answer "did we visit a column reference?"